--- a/window/inc/Window.h
+++ b/window/inc/Window.h
@@ -614,7 +614,8 @@
 	static LONG APIENTRY StaticWndProc( HWND hWnd, UINT Message, UINT wParam, LONG lParam )
 	{
 		guard(WWindow::StaticProc);
-		for( INT i=0; i<_Windows.Num(); i++ )
+		INT i;
+		for( i=0; i<_Windows.Num(); i++ )
 			if( _Windows(i)->hWnd==hWnd )
 				break;
 		if( i==_Windows.Num() && (Message==WM_NCCREATE || Message==WM_INITDIALOG) )
@@ -1585,7 +1586,7 @@
 	WUrlButton()
 	{}
 	WUrlButton( WWindow* InOwner, const TCHAR* InURL, INT InId=0 )
-	: WCoolButton( InOwner, InId, FDelegate(this,(TDelegate)OnClick) )
+	: WCoolButton( InOwner, InId, FDelegate(this,(TDelegate)&WUrlButton::OnClick) )
 	, URL( InURL )
 	{
 		FrameFlags = CBFF_ShowOver | CBFF_UrlStyle | CBFF_NoCenter;
@@ -2182,7 +2183,8 @@
 					Reader->Serialize( &AnsiText(0), AnsiText.Num() );
 					delete Reader;
 					INT CrCount=0;
-					for( INT Ofs=AnsiText.Num()-1; Ofs>0 && CrCount<MaxLines; Ofs-- )
+					INT Ofs;
+					for( Ofs=AnsiText.Num()-1; Ofs>0 && CrCount<MaxLines; Ofs-- )
 						CrCount += (AnsiText(Ofs)=='\n');
 					while( Ofs<AnsiText.Num() && AnsiText(Ofs)=='\n' )
 						Ofs++;
@@ -2278,7 +2280,7 @@
 			Exec->Exec( TEXT("TakeFocus"), *GLogWindow );
 			TCHAR NewURL[1024];
 			if
-			(	ParseToken(*(TCHAR**)&CD->lpData,NewURL,ARRAY_COUNT(NewURL),0)
+			(	ParseToken(*(const TCHAR**)&CD->lpData,NewURL,ARRAY_COUNT(NewURL),0)
 			&&	NewURL[0]!='-')
 				Exec->Exec( *(US+TEXT("Open ")+NewURL),*GLogWindow );
 		}
@@ -2425,7 +2427,8 @@
 	{
 		guard(WDialog::OnInitDialog);
 		WWindow::OnInitDialog();
-		for( INT i=0; i<Controls.Num(); i++ )
+		INT i;
+		for( i=0; i<Controls.Num(); i++ )
 		{
 			// Bind all child controls.
 			WControl* Control = Controls(i);
@@ -2485,8 +2488,8 @@
 	// Constructor.
 	WPasswordDialog()
 	: WDialog	  ( TEXT("PasswordDialog"), IDDIALOG_Password )
-	, OkButton    ( this, IDOK,     FDelegate(this,(TDelegate)EndDialogTrue) )
-	, CancelButton( this, IDCANCEL, FDelegate(this,(TDelegate)EndDialogFalse) )
+	, OkButton    ( this, IDOK,     FDelegate(this,(TDelegate)&WDialog::EndDialogTrue) )
+	, CancelButton( this, IDCANCEL, FDelegate(this,(TDelegate)&WDialog::EndDialogFalse) )
 	, Name		  ( this, IDEDIT_Name )
 	, Password	  ( this, IDEDIT_Password )
 	, Prompt      ( this, IDLABEL_Prompt )
@@ -2535,7 +2538,7 @@
 	WTextScrollerDialog( const TCHAR* InCaption, const TCHAR* InMessage )
 	: WDialog	( TEXT("TextScrollerDialog"), IDDIALOG_TextScroller )
 	, TextEdit  ( this, IDEDIT_TextEdit )
-	, OkButton  ( this, IDOK, FDelegate(this,(TDelegate)EndDialogTrue) )
+	, OkButton  ( this, IDOK, FDelegate(this,(TDelegate)&WDialog::EndDialogTrue) )
 	, Caption   ( InCaption )
 	, Message   ( InMessage )
 	{}
@@ -3327,7 +3330,8 @@
 		while( Index+1<Count )
 		{
 			FTreeItem* NextItem = OwnerProperties->GetListItem(Index+1);
-			for( FTreeItem* Check=NextItem->Parent; Check && Check!=this; Check=Check->Parent );
+			FTreeItem* Check;
+			for( Check=NextItem->Parent; Check && Check!=this; Check=Check->Parent );
 			if( !Check )
 				break;
 			NextItem->Expanded = 0;
@@ -3648,8 +3652,8 @@
 				Rect.Min.X = 28+OwnerProperties->GetDividerWidth();
 				Rect.Min.Y--;
 				TrackControl = new WTrackBar( &OwnerProperties->List );
-				TrackControl->ThumbTrackDelegate    = FDelegate(this,(TDelegate)OnTrackBarThumbTrack);
-				TrackControl->ThumbPositionDelegate = FDelegate(this,(TDelegate)OnTrackBarThumbPosition);
+				TrackControl->ThumbTrackDelegate    = FDelegate(this,(TDelegate)&FPropertyItem::OnTrackBarThumbTrack);
+				TrackControl->ThumbPositionDelegate = FDelegate(this,(TDelegate)&FPropertyItem::OnTrackBarThumbPosition);
 				TrackControl->OpenWindow( 0 );
 				TrackControl->SetTicFreq( 32 );
 				TrackControl->SetRange( 0, 255 );
@@ -3675,8 +3679,8 @@
 
 				ComboControl = new WComboBox( HolderControl );
 				ComboControl->Snoop = this;
-				ComboControl->SelectionEndOkDelegate     = FDelegate(this,(TDelegate)ComboSelectionEndOk);
-				ComboControl->SelectionEndCancelDelegate = FDelegate(this,(TDelegate)ComboSelectionEndCancel);
+				ComboControl->SelectionEndOkDelegate     = FDelegate(this,(TDelegate)&FPropertyItem::ComboSelectionEndOk);
+				ComboControl->SelectionEndCancelDelegate = FDelegate(this,(TDelegate)&FPropertyItem::ComboSelectionEndCancel);
 				ComboControl->OpenWindow( 0 );
 				ComboControl->MoveWindow( Rect-HolderRect.Min, 1 );
 
@@ -3718,29 +3722,29 @@
 			{
 				if( Expandable )
 				{
-					AddButton( LocalizeGeneral("AddButton",TEXT("Window")), FDelegate(this,(TDelegate)OnArrayAdd) );
+					AddButton( LocalizeGeneral("AddButton",TEXT("Window")), FDelegate(this,(TDelegate)&FPropertyItem::OnArrayAdd) );
 				}
-				AddButton( LocalizeGeneral("EmptyButton",TEXT("Window")), FDelegate(this,(TDelegate)OnArrayEmpty) );
+				AddButton( LocalizeGeneral("EmptyButton",TEXT("Window")), FDelegate(this,(TDelegate)&FPropertyItem::OnArrayEmpty) );
 			}
 			if( Cast<UArrayProperty>(Property->GetOuter()) )
 			{
 				if( Parent->Expandable )
 				{
-					AddButton( LocalizeGeneral("InsertButton",TEXT("Window")), FDelegate(this,(TDelegate)OnArrayInsert) );
-					AddButton( LocalizeGeneral("DeleteButton",TEXT("Window")), FDelegate(this,(TDelegate)OnArrayDelete) );
+					AddButton( LocalizeGeneral("InsertButton",TEXT("Window")), FDelegate(this,(TDelegate)&FPropertyItem::OnArrayInsert) );
+					AddButton( LocalizeGeneral("DeleteButton",TEXT("Window")), FDelegate(this,(TDelegate)&FPropertyItem::OnArrayDelete) );
 				}
 			}
 			if( Property->IsA(UStructProperty::StaticClass()) && appStricmp(Cast<UStructProperty>(Property)->Struct->GetName(),TEXT("Color"))==0 )
 			{
 				// Color.
-				AddButton( LocalizeGeneral("BrowseButton",TEXT("Window")), FDelegate(this,(TDelegate)OnChooseColorButton) );
+				AddButton( LocalizeGeneral("BrowseButton",TEXT("Window")), FDelegate(this,(TDelegate)&FPropertyItem::OnChooseColorButton) );
 			}
 			else if( Property->IsA(UObjectProperty::StaticClass()) )
 			{
 				// Class.
-				AddButton( LocalizeGeneral("BrowseButton",TEXT("Window")), FDelegate(this,(TDelegate)OnBrowseButton) );
-				AddButton( LocalizeGeneral("UseButton",   TEXT("Window")), FDelegate(this,(TDelegate)OnUseCurrentButton) );
-				AddButton( LocalizeGeneral("ClearButton", TEXT("Window")), FDelegate(this,(TDelegate)OnClearButton) );
+				AddButton( LocalizeGeneral("BrowseButton",TEXT("Window")), FDelegate(this,(TDelegate)&FPropertyItem::OnBrowseButton) );
+				AddButton( LocalizeGeneral("UseButton",   TEXT("Window")), FDelegate(this,(TDelegate)&FPropertyItem::OnUseCurrentButton) );
+				AddButton( LocalizeGeneral("ClearButton", TEXT("Window")), FDelegate(this,(TDelegate)&FPropertyItem::OnClearButton) );
 			}
 			if
 			(	(Property->IsA(UFloatProperty ::StaticClass()))
@@ -4196,8 +4200,8 @@
 		if( PersistentName!=NAME_None )
 			GConfig->GetInt( TEXT("WindowPositions"), *(FString(*PersistentName)+TEXT(".Split")), DividerWidth );
 		PropertiesWindows.AddItem( this );
-		List.DoubleClickDelegate     = FDelegate(this,(TDelegate)OnListDoubleClick);
-		List.SelectionChangeDelegate = FDelegate(this,(TDelegate)OnListSelectionChange);
+		List.DoubleClickDelegate     = FDelegate(this,(TDelegate)&WProperties::OnListDoubleClick);
+		List.SelectionChangeDelegate = FDelegate(this,(TDelegate)&WProperties::OnListSelectionChange);
 		unguard;
 	}
 
@@ -4465,7 +4469,8 @@
 			QWORD      Id   = Item->GetId();
 			if( Item->Expandable && !Item->Expanded )
 			{
-				for( INT j=0; j<Remembered.Num(); j++ )
+				INT j;
+				for( j=0; j<Remembered.Num(); j++ )
 					if( Remembered(j)==Id )
 						break;
 				if( j<Remembered.Num() )
@@ -4891,7 +4896,7 @@
 	void OnItemSetFocus()
 	{
 		FPropertyItemBase::OnItemSetFocus();
-		AddButton( LocalizeGeneral("DefaultsButton",TEXT("Window")), FDelegate(this,(TDelegate)OnResetToDefaultsButton) );
+		AddButton( LocalizeGeneral("DefaultsButton",TEXT("Window")), FDelegate(this,(TDelegate)&FObjectConfigItem::OnResetToDefaultsButton) );
 	}
 	void Expand()
 	{
@@ -4982,7 +4987,8 @@
 		UObject::GetPreferences( NewPrefs, *Prefs.Caption, 0 );
 		for( INT i=0; i<NewPrefs.Num(); i++ )
 		{
-			for( INT j=0; j<Children.Num(); j++ )
+			INT j;
+			for( j=0; j<Children.Num(); j++ )
 			{
 				if( appStricmp( *Children(j)->GetCaption(), *NewPrefs(i).Caption )==0 )
 					break;
@@ -5114,10 +5120,10 @@
 	WWizardDialog()
 	: WDialog	    ( TEXT("WizardDialog"), IDDIALOG_WizardDialog )
 	, PageHolder    ( this, IDC_PageHolder )
-	, BackButton    ( this, IDC_Back,   FDelegate(this,(TDelegate)OnBack  ) )
-	, NextButton    ( this, IDC_Next,   FDelegate(this,(TDelegate)OnNext  ) )
-	, FinishButton  ( this, IDC_Finish, FDelegate(this,(TDelegate)OnFinish) )
-	, CancelButton  ( this, IDCANCEL,   FDelegate(this,(TDelegate)OnCancel) )
+	, BackButton    ( this, IDC_Back,   FDelegate(this,(TDelegate)&WWizardDialog::OnBack  ) )
+	, NextButton    ( this, IDC_Next,   FDelegate(this,(TDelegate)&WWizardDialog::OnNext  ) )
+	, FinishButton  ( this, IDC_Finish, FDelegate(this,(TDelegate)&WWizardDialog::OnFinish) )
+	, CancelButton  ( this, IDCANCEL,   FDelegate(this,(TDelegate)&WWizardDialog::OnCancel) )
 	, Pages         ()
 	, CurrentPage   ( NULL )
 	{}
